File: Iteration2/code/CSE3063GRP12/src/features/student/course_registration/CourseRegistrationView.java
package features.student.course_registration;

import java.util.ArrayList;

import core.enums.ApprovalState;
import core.models.concretes.Course;

public class CourseRegistrationView extends Exception {
	public void showCourseList(ArrayList<Course> courseList) {
		System.out.println("=======================================================||Course Registration||=======================================================");
		System.out.println("###======List of Available Courses:");
		System.out.println("=====================================================================================================================================");
		System.out.printf("%-3s\t%-20s %-50s %-8s %-26s %-8s %-8s\n", " No.", "Course Code", "Course Name", "Quota", "Registered Students No.", "Semester No.", "ECTS");
		System.out.println("=====================================================================================================================================");
		for (int i = 0; i < courseList.size(); i++) {
			System.out.printf(" [%-1d]\t%-20s %-52s %-16d %-22d %-8d %-8d\n", i + 1, courseList.get(i).getCourseCode(),
            courseList.get(i).getName(), courseList.get(i).getQuota(), courseList.get(i).getCurrentQuota(), courseList.get(i).getSemester() ,courseList.get(i).getCredit());
		}
		System.out.println("=====================================================================================================================================");
	}

	public void showCourseList(ArrayList<Course> courseList, ApprovalState approvalState) {
		/*
		 * If student sends his/her course list for approval and clicks again on the
		 * CourseRegistrationController,
		 * in the CourseEnrollmentRepository, there is a method
		 * getCourseEnrollmentByStudentId(String), we will get the CourseEnrollment
		 * model.
		 * Inside it, there are courseList, studentId, and approvalState. If the student
		 * didn't send anything to the advisor, then
		 * there will be no CourseEnrollment
		 * Must know whether the return type will be Null or CourseEnrollment
		 * CourseEnrollment courseEnrollment =
		 * courseRegistrationRepositry.getCourseEnrollmentByStudentId()
		 * if ce is true, courseEnrollment.getSelectedCourseList(),
		 * courseEnrollment.getApprovalState()
		 * print the courseList and printf("Your approval status is %s, ApprovalState)
		 * add, if you want to go back, press q
		 */
		System.out.println("Your list of courses for this semester:");
		for (int i = 0; i < courseList.size(); i++) {
			System.out.println((i + 1) + ". " + courseList.get(i).getName());
		}
		System.out.printf("Your registration status is: %s \n\n", approvalState);

	}

	public void showSuccessMessage() {
		System.out.println("Your enrollment has been saved successfully");
	}

	public void showErrorMessage(Exception exception) {
		System.err.println(exception.getMessage());
	}
}

File: Iteration3/code/CSE3063GRP12/src/features/student/course_registration/CourseRegistrationView.py
class CourseRegistrationView:

    def show_course_list(self, *args):
        # args[0] is the list of courses
        # args[1] is the approval state
        if len(args) == 1:
            print(f"{'No.':<3}\t{'Course Code':<20} {'Course Name':<50} {'Quota':<8} {'Current Quota':<15} {'Semester No.':<8} {'ECTS':<8}")
            print("=" * 128)
            for i, course in enumerate(args[0]):
                if isinstance(course, dict):
                    # Accessing dictionary keys
                    course_code = course.get('courseCode')
                    name = course.get('name')
                    quota = course.get('quota')
                    current_quota = course.get('currentQuota')
                    semester = course.get('semester')
                    credit = course.get('credit')
                else:
                    # Accessing attributes of Course object
                    course_code = course.courseCode
                    name = course.name
                    quota = course.quota
                    current_quota = course.currentQuota
                    semester = course.semester
                    credit = course.credit
                
                print(f"[{i + 1:<1}]\t{course_code:<20} {name:<50} {quota:<8} {current_quota:<15} {semester:<8} {credit:<8}")
            print("=" * 128)
        elif len(args) == 2:
            print("Your list of courses for this semester:")
            for i, course in enumerate(args[0]):
                print(f"{i + 1}. {course.get('name')}")
            print(f"Your registration status is: {args[1]} \n\n")
        else:
            print("Error: Invalid arguments passed to show_course_list()")

    def show_success_message(self):
        print("Your enrollment has been saved successfully")
    
    def show_error_message(self, exception):
        print(f"An error occurred while saving your enrollment: {exception}")